Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java

```diff
@@ -0,0 +1,447 @@
+    private static final String MODULE_NAME = ILeappAnalyzerModuleFactory.getModuleName();
+
+    private static final String XMLFILE = "ileap-artifact-attribute-reference.xml"; //NON-NLS
+
+    private final Map<String, String> tsvFiles;
+    private final Map<String, String> tsvFileArtifacts;
+    private final Map<String, String> tsvFileArtifactComments;
+    private final Map<String, List<List<String>>> tsvFileAttributes;
+
+    public ILeappFileProcessor() throws IOException, IngestModuleException {
+        this.tsvFiles = new HashMap<>();
+        this.tsvFileArtifacts = new HashMap<>();
+        this.tsvFileArtifactComments = new HashMap<>();
+        this.tsvFileAttributes = new HashMap<>();
+
+        configExtractor();
+        loadConfigFile();
+
+    }
+
+    @NbBundle.Messages({
+        "ILeappFileProcessor.error.running.iLeapp=Error running iLeapp, see log file.",
+        "ILeappFileProcessor.error.creating.output.dir=Error creating iLeapp module output directory.",
+        "ILeappFileProcessor.starting.iLeapp=Starting iLeapp",
+        "ILeappFileProcessor.running.iLeapp=Running iLeapp",
+        "ILeappFileProcessor.has.run=iLeapp",
+        "ILeappFileProcessor.iLeapp.cancelled=iLeapp run was canceled",
+        "ILeappFileProcessor.completed=iLeapp Processing Completed",
+        "ILeappFileProcessor.error.reading.iLeapp.directory=Error reading iLeapp Output Directory"})
+
+    public ProcessResult processFiles(Content dataSource, Path moduleOutputPath, AbstractFile iLeappFile) {
+
+        try {
+            List<String> iLeappTsvOutputFiles = findTsvFiles(moduleOutputPath);
+            if (!iLeappTsvOutputFiles.isEmpty()) {
+                processiLeappFiles(iLeappTsvOutputFiles, iLeappFile);
+            }
+        } catch (IOException | IngestModuleException ex) {
+            logger.log(Level.SEVERE, String.format("Error trying to process iLeapp output files in directory %s. ", moduleOutputPath.toString()), ex); //NON-NLS
+            return ProcessResult.ERROR;
+        }
+
+        return ProcessResult.OK;
+    }
+
+    /**
+     * Find the tsv files in the iLeapp output directory and match them to files
+     * we know we want to process and return the list to process those files.
+     */
+    private List<String> findTsvFiles(Path iLeappOutputDir) throws IngestModuleException {
+        List<String> allTsvFiles = new ArrayList<>();
+        List<String> foundTsvFiles = new ArrayList<>();
+
+        try (Stream<Path> walk = Files.walk(iLeappOutputDir)) {
+
+            allTsvFiles = walk.map(x -> x.toString())
+                    .filter(f -> f.toLowerCase().endsWith(".tsv")).collect(Collectors.toList());
+
+            for (String tsvFile : allTsvFiles) {
+                if (tsvFiles.containsKey(FilenameUtils.getName(tsvFile))) {
+                    foundTsvFiles.add(tsvFile);
+                }
+            }
+
+        } catch (IOException e) {
```

Comment:
See previous comments on Files.walk() in the ingest module, UncheckedIOException should be caught here too.

Reply:
Added check for exception